release-to-candidate/action.yaml

```
@@ -46,21 +52,52 @@ runs:
         # Install moreutils so we have access to sponge
         sudo apt-get update; sudo apt-get install -y moreutils
 
+    - name: Find the snapcraft.yaml path
+      id: yaml-path
+      shell: bash
+      run : |
+        if [ -n "${{ inputs.yaml-path }}" ]; then
+            yaml_path="${{ inputs.yaml-path }}"
+        elif [ -e snap/snapcraft.yaml ]; then
+            yaml_path="snap/snapcraft.yaml"
+        elif [ -e snapcraft.yaml ]; then
+            yaml_path="snapcraft.yaml"
+        elif [ -e build-aux/snap/snapcraft.yaml ]; then
+            yaml_path="build_aux/snap/snapcraft.yaml"
+        elif [ -e .snapcraft.yaml ]; then
+            yaml_path=.snapcraft.yaml
+        else
+            echo "Could not find snapcraft.yaml"
+            exit 1
+        fi
+        echo "yaml_path=${yaml_path}" >> "$GITHUB_OUTPUT"
+
+
     - name: Build the snap (${{ inputs.architecture }})
       id: build
       shell: bash
+      with:
+        yaml_path: ${{ steps.yaml-path.outputs.yaml_path }}
       env:
-        name: ${{ github.event.repository.name }}
+        name: ${{ if not(inputs.name) }}${{ github.event.repository.name }}${{ else }}${{ inputs.name }}${{ endif }} 
```

```suggestion
        name: ${{ inputs.name || github.event.repository.name }} 
```

I think this should work and is quite a lot more succinct